src/custom-hooks/useFetch.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+
+import { TimelineData } from "../mocks/timeLine.data.mock";
+
+interface Defaults {
+  loading: boolean;
+  error: any;
+  data: any;
+}
+
+let defaults: Defaults = {
+  loading: false,
+  error: undefined,
+  data: undefined
+};
+
+const useFetch = (url: string) => {
+  let [loading, setLoading] = useState(defaults.loading);
+  let [error, setError] = useState(defaults.error);
+  let [data, setData] = useState(defaults.data);
+
+  const setToLoading = (): void => {
+    setLoading(true);
+    setData(undefined);
+    setError(undefined);
+  };
+
+  const setToData = (data: any) => {
+    setLoading(false);
+    setData(data);
+    setError(undefined);
+  };
+
+  const setToError = (errorMessage: any) => {
+    setLoading(false);
+    setData(undefined);
+    setError(errorMessage);
+  };
+
+  useEffect(() => {
+    const fetchUrl = async () => {
+      try {
+        setToLoading();
```

Comment:
We ideally want to show/hide loading automatically, based on state of our data. Manual setup/teardown causes edge-cases when things get complex.
We can let this be for now, but will improve in future as a small PR